### ISSUE TITLE
pyzmq 21 shared object 12 times larger than in v20

### ISSUE THREAD

**Issue by Jeducious** [zeromq/pyzmq#1491]
Hi there, so I recently upgraded to version 21 of pyzmq, things were a little different I noticed. 

<img width="690" alt="Screen Shot 2021-01-28 at 10 43 01 am" src="https://user-images.githubusercontent.com/34228929/106068954-9c759c80-6155-11eb-899e-6e4145ba61bd.png">

As you can see in the screenshot, the libzmq so is only 975kB for python 3.8 but is 12MB for python 3.7

All I did was do a pip install like this

```python
    proc = subprocess.Popen(
                    [sys.executable, '-m',
                     'pip', 'install',
                     '--target', parent_dir,
                     '--platform', platform_tag, 
                     #replace underscores as pip uses "37" not "3_7" for 
                     # major and minor versions of python
                     '--python-version', py_version.replace("_","",-1),
                     #default upgrade strategy is 'only_if_needed'
                     # see https://pip.pypa.io/en/stable/reference/pip_install/#options
                     '--upgrade',
                     '--progress-bar', 'pretty',
                     '--ignore-installed',
                     '--no-deps',
                     dep_name])
```

Other platforms were unaffected. 

Is the libzmq supposed to be 12 times larger for 3.7 vs 3.8? Seems unlikely xD

**Comment by minrk** [comment]
Yup! The migration from multibuild to cibuildwheel in v21 meant losing `-Wl,-strip-all` in the default CFLAGS, which I hadn't noticed. Fixed by #1493. Thanks for checking in!

**Comment by Jeducious** [comment]
No problem :) Yeah, it looked a little odd, glad it helped.
